cpp/deplex/src/deplex/cell_segment.cpp

```diff
@@ -71,11 +60,19 @@ bool CellSegment::isVerticalContinuous(Eigen::MatrixXf const& cell_z) const {
   return disc_count < config_->getInt("maxNumberDepthDiscontinuity");
 }
 
-bool CellSegment::isDepthContinuous() const {
-  Eigen::MatrixXf cell_z = ptr_pcd_array_->block(offset_, 2, nr_pts_per_cell_, 1).reshaped(cell_height_, cell_width_);
+bool CellSegment::isDepthContinuous(Eigen::MatrixXf const& cell_points) const {
+  auto cell_height = config_->getInt("patchSize");
+  auto cell_width = config_->getInt("patchSize");
+  Eigen::MatrixXf cell_z = cell_points.col(2).reshaped(cell_height, cell_width);
 
   return isHorizontalContinuous(cell_z) && isVerticalContinuous(cell_z);
 }
 
-void CellSegment::calculateStats() { stats_.fitPlane(); }
+bool CellSegment::isFittingMSE() const {
+  if (stats_.getMSE() < 0) return false;
+  float depth_sigma_coeff = config_->getFloat("depthSigmaCoeff");
```

Comment:
Let's use common way, above you have `auto` for such use case, why it is different here?